bsondump/bsondump_test.go

```diff
@@ -8,172 +8,284 @@ package bsondump
+	bufRef.ReadFrom(outReference)
+	bufRefStr := bufRef.String()
+
+	bufDumpStr := cmdOutput.String()
+	require.Equal(bufRefStr, bufDumpStr)
+}
+
+func testFromFileWithNamedArgumentToFile(t *testing.T) {
+	require := require.New(t)
+
+	dir, cleanup := testutil.MakeTempDir(t)
+	defer cleanup()
+	outFile := filepath.Join(dir, "out.json")
+
+	cmd := bsondumpCommand("--outFile", outFile, "--bsonFile", "testdata/sample.bson")
+
+	err := cmd.Run()
+	require.NoError(err)
+
+	// Get the correct bsondump result from a file to use as a reference.
+	outReference, err := os.Open("testdata/sample.json")
+	require.NoError(err)
+	bufRef := new(bytes.Buffer)
+	bufRef.ReadFrom(outReference)
+	bufRefStr := bufRef.String()
+
+	// Get the output from a file.
+	outDump, err := os.Open(outFile)
+	require.NoError(err)
+	bufDump := new(bytes.Buffer)
+	bufDump.ReadFrom(outDump)
+	bufDumpStr := bufDump.String()
+
+	require.Equal(bufRefStr, bufDumpStr)
+}
+
+func testFromFileWithPositionalArgumentToFile(t *testing.T) {
+	require := require.New(t)
 
-	Convey("Test bsondump reading from a file with a positional arg and writing to a file", t, func() {
-		cmd := exec.Command(executable, "--outFile", "out.json", "testdata/sample.bson")
+	dir, cleanup := testutil.MakeTempDir(t)
+	defer cleanup()
+	outFile := filepath.Join(dir, "out.json")
 
-		err := cmd.Run()
-		So(err, ShouldBeNil)
+	cmd := bsondumpCommand("--outFile", outFile, "testdata/sample.bson")
 
-		// Get the correct bsondump result from a file to use as a reference.
-		outReference, err := os.Open("testdata/sample.json")
-		So(err, ShouldBeNil)
-		bufRef := new(bytes.Buffer)
-		bufRef.ReadFrom(outReference)
-		bufRefStr := bufRef.String()
+	err := cmd.Run()
+	require.NoError(err)
 
-		// Get the output from a file.
-		outDump, err := os.Open("out.json")
-		So(err, ShouldBeNil)
-		bufDump := new(bytes.Buffer)
-		bufDump.ReadFrom(outDump)
-		bufDumpStr := bufDump.String()
+	// Get the correct bsondump result from a file to use as a reference.
+	outReference, err := os.Open("testdata/sample.json")
+	require.NoError(err)
+	bufRef := new(bytes.Buffer)
+	bufRef.ReadFrom(outReference)
+	bufRefStr := bufRef.String()
 
-		So(bufDumpStr, ShouldEqual, bufRefStr)
+	// Get the output from a file.
+	outDump, err := os.Open(outFile)
+	require.NoError(err)
+	bufDump := new(bytes.Buffer)
+	bufDump.ReadFrom(outDump)
+	bufDumpStr := bufDump.String()
 
-		Reset(func() {
-			os.Remove("out.json")
-		})
+	require.Equal(bufRefStr, bufDumpStr)
+}
+
+func bsondumpCommand(args ...string) *exec.Cmd {
+	cmd := []string{"go", "run", filepath.Join("..", "bsondump", "main")}
+	cmd = append(cmd, args...)
+	return exec.Command(cmd[0], cmd[1:]...)
+}
+
+func TestBsondumpMaxBSONSize(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+
+	// 16mb
+	size := int(16*math.Pow(1024, 2)) -
+		// subtract 5 for the 4 bytes for the int32 at the head of the document that
+		// specifies its size and 1 for its trailing NULL.
+		5 -
```

Comment:
Hmm I'm not a huge fan of this formatting. I think it would be much cleaner to add together all the subtractions and have one big block comment explaining how to construct the number. Or at least group together on one line. The BSON spec is very stable so it's not like this is something we expect to need to edit.